TestingLowerBounds/MeasureCompProd.lean

```diff
@@ -73,10 +94,22 @@ lemma kernel.snd_comp {γ δ : Type*} {_ : MeasurableSpace γ} {_ : MeasurableSp
 Defined using `MeasureTheory.Measure.bind` -/
 scoped[ProbabilityTheory] infixl:100 " ∘ₘ " => MeasureTheory.Measure.bind
 
-lemma Measure.fst_compProd (μ : Measure α) [SFinite μ] (κ : kernel α β) [IsMarkovKernel κ] :
-    (μ ⊗ₘ κ).fst = μ := by
-  ext s
-  rw [Measure.compProd, Measure.fst, ← kernel.fst_apply, kernel.fst_compProd, kernel.const_apply]
+lemma Measure.comp_assoc {γ : Type*} {_ : MeasurableSpace γ} {μ : Measure α} [SFinite μ]
+    {κ : kernel α β} [IsSFiniteKernel κ]
+    {η : kernel β γ} [IsFiniteKernel η] :
+    μ ∘ₘ κ ∘ₘ η = μ ∘ₘ (η ∘ₖ κ) :=
+  Measure.bind_bind (kernel.measurable _) (kernel.measurable _)
+
+lemma Measure.comp_deterministic_eq_map {f : α → β} (hf : Measurable f) :
+    μ ∘ₘ kernel.deterministic f hf = μ.map f := by
+  ext s hs
+  rw [Measure.bind_apply hs (kernel.measurable _), Measure.map_apply hf hs]
+  simp only [kernel.deterministic_apply' hf _ hs]
+  classical
+  rw [lintegral_indicator_const_comp hf hs, one_mul]
```

Comment:
```suggestion
  simp only [kernel.deterministic_apply' hf _ hs]
  rw [lintegral_indicator_const_comp hf hs, one_mul]
```

This seems to compile even without the `classical`.

Reply:
Oh right. I added `classical` because I was using `indicator_apply` but then I changed the proof and that was not needed anymore.
Thanks!